server/utils/stigLibrary.ts

```diff
@@ -0,0 +1,291 @@
+import * as fs from "fs";
+import * as path from "path";
+import AdmZip from "adm-zip";
+import { DateTime } from "luxon";
+import { Op, UniqueConstraintError } from "sequelize";
+import {
+  Assessment,
+  AssessmentItem,
+  StigData,
+  StigLibrary,
+  System,
+  Stig,
+  StigAlias,
+} from "../../db/models";
+import { hashFile } from "./hash";
+import { createBlankAssessment } from "./assessments";
+
+export type ProcessLibraryResults = {
+  stigProcessed: number;
+  newStigCount: number;
+  updatedStigCount: number;
+  unchangedStigCount: number;
+  xmlsExtracted: number;
+};
+
+export const migrateBoundary = async (
+  boundaryId: number,
+  newStigLibraryId: number,
+): Promise<{ results: string }> => {
+  try {
+    const boundarySystems = await System.findAll({ where: { BoundaryId: boundaryId } });
+    for (const system of boundarySystems) {
+      const currentAssessments = await Assessment.findAll({
+        where: {
+          SystemId: system.id,
+          succeededByAssessmentId: { [Op.is]: null },
+        },
+      });
+      for (const assessment of currentAssessments) {
+        const oldStig = await Stig.findOne({ where: { id: assessment.StigId } });
```

Comment:
would help if we could self document / just actually straight up document that assessment.stigid is like the db id and then oldstig.stigid is the id coming from the stig itself (firefox blah blah)